core/src/impl/Kokkos_Core.cpp

```diff
@@ -474,42 +483,62 @@ void parse_command_line_arguments(int& narg, char* arg[],
   int iarg = 0;
 
   while (iarg < narg) {
-    if (Impl::check_int_arg(arg[iarg], "--kokkos-threads", &num_threads)) {
+    if (check_int_arg(arg[iarg], "--kokkos-threads", &num_threads)) {
       for (int k = iarg; k < narg - 1; k++) {
         arg[k] = arg[k + 1];
       }
       kokkos_threads_found = 1;
       narg--;
     } else if (!kokkos_threads_found &&
-               Impl::check_int_arg(arg[iarg], "--threads", &num_threads)) {
+               check_int_arg(arg[iarg], "--threads", &num_threads)) {
       iarg++;
-    } else if (Impl::check_int_arg(arg[iarg], "--kokkos-numa", &numa)) {
+    } else if (check_int_arg(arg[iarg], "--kokkos-numa", &numa)) {
       for (int k = iarg; k < narg - 1; k++) {
         arg[k] = arg[k + 1];
       }
       kokkos_numa_found = 1;
       narg--;
     } else if (!kokkos_numa_found &&
-               Impl::check_int_arg(arg[iarg], "--numa", &numa)) {
+               check_int_arg(arg[iarg], "--numa", &numa)) {
       iarg++;
-    } else if (Impl::check_int_arg(arg[iarg], "--kokkos-device", &device)) {
+    } else if (check_int_arg(arg[iarg], "--kokkos-device-id", &device) ||
+               check_int_arg(arg[iarg], "--kokkos-device", &device)) {
+      if (check_arg(arg[iarg], "--kokkos-device")) {
+        warn_deprecated_command_line_argument("--kokkos-device",
+                                              "--kokkos-device-id");
+      }
       for (int k = iarg; k < narg - 1; k++) {
         arg[k] = arg[k + 1];
       }
       kokkos_device_found = 1;
       narg--;
     } else if (!kokkos_device_found &&
-               Impl::check_int_arg(arg[iarg], "--device", &device)) {
+               (check_int_arg(arg[iarg], "--device-id", &device) ||
+                check_int_arg(arg[iarg], "--device", &device))) {
+      if (check_arg(arg[iarg], "--device")) {
+        warn_deprecated_command_line_argument("--device", "--device-id");
+      }
       iarg++;
-    } else if (Impl::check_arg(arg[iarg], "--kokkos-ndevices") ||
-               Impl::check_arg(arg[iarg], "--ndevices")) {
+    } else if (check_arg(arg[iarg], "--kokkos-num-devices") ||
+               check_arg(arg[iarg], "--num-devices") ||
+               check_arg(arg[iarg], "--kokkos-ndevices") ||
+               check_arg(arg[iarg], "--ndevices")) {
+      if (check_arg(arg[iarg], "--ndevices")) {
+        warn_deprecated_command_line_argument("--ndevices", "--num-devices");
+      }
+      if (check_arg(arg[iarg], "--kokkos-ndevices")) {
+        warn_deprecated_command_line_argument("--kokkos-ndevices",
+                                              "--kokkos-num-devices");
+      }
       // Find the number of device (expecting --device=XX)
-      if (!((strncmp(arg[iarg], "--kokkos-ndevices=", 18) == 0) ||
+      if (!((strncmp(arg[iarg], "--kokkos-num-devices=", 21) == 0) ||
```

Comment:
Any reason not to use `check_arg` here?

Reply:
I did not (and still don't) want to mess with what goes inside the statement-true of that outer if-else statement.  It has other problems but it is good enough and I don't want to rewrite it.